packages/headless-components/stores/src/react/SelectedVariant.tsx

```diff
@@ -1,6 +1,63 @@
 import type { ServiceAPI } from "@wix/services-definitions";
-import { useService } from "@wix/services-manager-react";
-import { SelectedVariantServiceDefinition } from "../services/selected-variant-service.js";
+import { useService, WixServices } from "@wix/services-manager-react";
+import {
+  SelectedVariantServiceDefinition,
+  SelectedVariantService,
+  SelectedVariantServiceConfig,
+} from "../services/selected-variant-service.js";
+import { createServicesMap } from "@wix/services-manager";
+import type { PropsWithChildren } from "react";
+
+/**
+ * Root component that provides the SelectedVariant service context to its children.
+ * This component sets up the necessary services for rendering and managing selected variant data.
+ *
+ * @order 1
+ * @component
+ * @example
+ * ```tsx
+ * import { SelectedVariant } from '@wix/stores/components';
+ *
+ * function ProductVariantDisplay() {
+ *   return (
+ *     <SelectedVariant.Root selectedVariantServiceConfig={{ fetchInventoryData: true }}>
+ *       <div>
+ *         <SelectedVariant.Price>
+ *           {({ price, compareAtPrice, currency }) => (
+ *             <div className="price-display">
+ *               <span className="current-price">{price}</span>
+ *               {compareAtPrice && (
+ *                 <span className="compare-price">
+ *                   <s>{compareAtPrice}</s>
+ *                 </span>
+ *               )}
+ *               <span className="currency">{currency}</span>
+ *             </div>
+ *           )}
+ *         </SelectedVariant.Price>
+ *       </div>
+ *     </SelectedVariant.Root>
+ *   );
+ * }
+ * ```
+ */
+export function Root(
+  props: PropsWithChildren<{
+    selectedVariantServiceConfig: SelectedVariantServiceConfig;
```

Comment:
Let's make the config optional:
```suggestion
    selectedVariantServiceConfig?: SelectedVariantServiceConfig = {};
```
So the user could use it more easily when there is no config:
```
<SelectedVariant.Root>
```